src/main/java/net/oauth/jsontoken/JsonToken.java

```diff
@@ -51,10 +51,10 @@
   
   public final static int DEFAULT_LIFETIME_IN_MINS = 2;
 
-
   private JsonObject header;
+  private String keyId;
   private SignatureAlgorithm sigAlg;
```

Comment:
These mutable variables seem unnecessary and make the code harder to understand. It looks like they never actually change either (I don't see any methods that could modify the header values). Is it possible to make them final, and set the values in the constructors?

Reply:
That's a good point, I didn't realize the header could be changed directly. It seems like a bug that getSignatureAlgorithm() caches the initial value if the caller is allowed to change it. Your approach SGTM.

I think we can still make the header object final and build that in the constructor.

Reply:
Header object is now final and is initialized in the constructor